### PR TITLE
feat(CareerPathCard): :sparkles: Adds salary metadata to CareerPathCard component

### PR DESCRIPTION
**Pull request by saghdaey**
### Overview

Adds salary information to career path cards when data is provided

### PR Checklist

- [X] Related to designs:
- [X] Related to JIRA ticket: [DISC-180]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories 

**I'm seeing a bunch of unrelated axe e2e tests fail for duplicate Ids (in mono); will look into this monday**
monolith tests, however, are all passing

#### Testing Instructions

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

Visit the [CareerPathCard story in the preview environment](https://64023aa09f540d03450eface--gamut-preview.netlify.app/), specifically the "CareerPathCard with Salary metadata" example 
Screen Shot 2023-03-03 at 3 14 55 PM

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/34895)  | [Monolith Env](https://pr-34895-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/1396)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-1396)   |

[DISC-180]: https://codecademy.atlassian.net/browse/DISC-180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.288-alpha.78d6b8.0<br/>@codecademy/gamut-labs@35.6.3-alpha.78d6b8.0<br/>@codecademy/styleguide@63.5.13-alpha.78d6b8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64023aa09f540d03450eface--gamut-preview.netlify.app>https://64023aa09f540d03450eface--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64023aa09f540d03450eface>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.289-alpha.b243e0.0<br/>@codecademy/gamut-labs@35.6.4-alpha.b243e0.0<br/>@codecademy/styleguide@63.5.14-alpha.b243e0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64024ce92d987c0d3a679e68--gamut-preview.netlify.app>https://64024ce92d987c0d3a679e68--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64024ce92d987c0d3a679e68>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.289-alpha.87ac16.0<br/>@codecademy/gamut-labs@35.6.4-alpha.87ac16.0<br/>@codecademy/styleguide@63.5.14-alpha.87ac16.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6402585b84f6a4006cb0e2c1--gamut-preview.netlify.app>https://6402585b84f6a4006cb0e2c1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6402585b84f6a4006cb0e2c1>Deploy Logs</a></p>

File: packages/gamut-labs/src/CareerPathCard/index.tsx [hunk 1]
```diff
@@ -47,6 +62,15 @@ export const CareerPathCard: React.FC<CareerPathCardProps> = ({
       isFullSize={isFullSize}
       enrollmentStatus={enrollmentStatus}
     >
+      {salary && (
```
Comment (saghdaey) [inline packages/gamut-labs/src/CareerPathCard/index.tsx:65]: salary data will only show if provided (gamut to have no knowledge of this business logic and just display if the data is there) 

consumer (mono, catalog pages) will handle logic of when to provide this data (catalog hub, catalog all, catalog home and ONLY when user is in the salary experiment variant group)

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6402638038068b2131690b50--gamut-preview.netlify.app>https://6402638038068b2131690b50--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6402638038068b2131690b50>Deploy Logs</a></p>

**Review by zippyzow — COMMENTED** [review]
Looking good in storybook! One question: Mark noted in the ticket, "course count is no longer displayed on career path card" (when salary is shown). Is this still a requirement?

**Comment by saghdaey** [comment]
> Looking good in storybook! One question: Mark noted in the ticket, "course count is no longer displayed on career path card" (when salary is shown). Is this still a requirement?

oops i missed that! in my head that was for time to complete, will remove

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.290-alpha.670b79.0<br/>@codecademy/gamut-labs@35.6.5-alpha.670b79.0<br/>@codecademy/styleguide@63.5.15-alpha.670b79.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64067e144667ef02747bb489--gamut-preview.netlify.app>https://64067e144667ef02747bb489--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64067e144667ef02747bb489>Deploy Logs</a></p>

File: packages/styleguide/stories/Brand Labs/Molecules/CareerPathCard.stories.mdx [hunk 2]
```diff
@@ -43,6 +43,19 @@ Note: This card will need to be wrapped in an Anchor or Link in order to be clic
   </Story>
 </Canvas>
 
+## CareerPathCard with Salary metadata
+
```
Comment (zippyzow) [inline packages/styleguide/stories/Brand Labs/Molecules/CareerPathCard.stories.mdx:47]:
```suggestion

Course count is hidden when salary is shown. 

```

**Review by zippyzow — APPROVED** [review]
Left small suggestion for storybook!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.290-alpha.6d1a0a.0<br/>@codecademy/gamut-labs@35.6.5-alpha.6d1a0a.0<br/>@codecademy/styleguide@63.5.15-alpha.6d1a0a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6407618921c7ff1d7585d32f--gamut-preview.netlify.app>https://6407618921c7ff1d7585d32f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6407618921c7ff1d7585d32f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://640768cf3dee31263b1eeb36--gamut-preview.netlify.app>https://640768cf3dee31263b1eeb36--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/640768cf3dee31263b1eeb36>Deploy Logs</a></p>

**Review by aresnik11 — APPROVED** [review]

**Pull request merged**